File: kotlin-pr/50/kotlin/code-quality-logging/ProfileUpdater.kt
class ProfileUpdater {
    fun updateProfile(userId: String, profileData: Map<String, String>): Boolean {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        println("Updating profile for user $userId")
        println("Profile data: $profileData")
        println("Email: ${profileData["email"]}")
        println("Phone: ${profileData["phone"]}")
        println("SSN: ${profileData["ssn"]}")
        
        // Update logic
        return true
        // {/fact}
    }
}

File: php-pr/51/php/code-quality-logging/ProfileUpdater.php
<?php

class ProfileUpdater
{
    public function updateProfile($userId, $profileData)
    {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        error_log("Updating profile for user {$userId}");
        error_log("Profile data: " . json_encode($profileData));
        error_log("Email: " . $profileData['email']);
        error_log("Phone: " . $profileData['phone']);
        error_log("SSN: " . $profileData['ssn']);
        
        // Update logic
        return true;
        // {/fact}
    }
}